llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/LlapDaemon.java

```diff
@@ -474,6 +478,10 @@ public void serviceStart() throws Exception {
       getConfig().setInt(ConfVars.LLAP_DAEMON_WEB_PORT.varname, webServices.getPort());
     }
     getConfig().setInt(ConfVars.LLAP_DAEMON_OUTPUT_SERVICE_PORT.varname, LlapOutputFormatService.get().getPort());
+    if (LlapUtil.isCloudDeployment()) {
+      getConfig().setInt(ConfVars.LLAP_EXTERNAL_CLIENT_CLOUD_RPC_PORT.varname,
```

Comment:
i think somewhere here we have to do a validation.. if the port is non-zero and if the jwt shared secret cannot be obtained then fail the daemon..

Reply:
done